src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/EventsPersister.cs

```diff
@@ -34,15 +34,21 @@ public async Task PersistNewMaturedBlockDepositsAsync(IMaturedBlockDeposits[] ma
         {
             this.logger.LogDebug("New {0} received.", nameof(IMaturedBlockDeposits));
 
-            for (int i = 0; i < maturedBlockDeposits.Length; i++)
+            if (maturedBlockDeposits?.Length > 0)
```

Comment:
is it possible to reduce the nesting a bit ?

Reply:
Done.